cli/index.ts

```diff
@@ -21,7 +21,7 @@ const errorOutput = (output: string) => coloredOutput(output, 91);
 const writeStdOut = (output: string) => process.stdout.write(output + "\n");
 const writeStdErr = (output: string) => process.stderr.write(output + "\n");
 
-const projectDirOption = {
+const projectDirOption: INamedOption<yargs.PositionalOptions> = {
```

Comment:
i would argue that in the general case, we shouldn't specify types on variable declarations. the great thing about typescript's type system (in particular interfaces) is that instead of having to declare tight dependencies (e.g. objects and the interfaces they implement), the binding is much looser so that it's much simpler for functions to specify the shape of the parameters they receive.

in this case, projectDirOption shouldn't have to care that it is any specific type - the place that it matters is the function that accepts the whole ICli object.

this is particularly powerful because a single object can implement multiple interfaces trivially without having to declare that it does so *until the place that it actually matters*.